libraries/radspreadprocessing/formats-and-conversion/import-and-export-to-excel-file-formats/xlsm/xlsmformatprovider.md

```diff
@@ -0,0 +1,102 @@
+---
+title: Using XlsmFormatProvider
+page_title: Import and Export to Xlsm File Format Using XlsmFormatProvider
+description: Import and Export to Xlsm File Format Using XlsmFormatProvider
+slug: radspreadprocessing-formats-and-conversion-xlsm-xlsmformatprovider
+tags: Xlsmformatprovider,excel,workbook,xlsm,import,export,save
+published: True
+position: 1
+
+---
+
+# Using XlsmFormatProvider
+
+
+__XlsmFormatProvider__ makes it easy to import and export Xlsm (Excel Workbook that supports Macros) files. An Xlsm file is a group of zipped files that conform to the Office Open XML schema. That said, the format allows you export all parts of a workbook: worksheets, formula values, formatting, hyperlinks, etc.
+
+> Currently the Macros are only preserved during import export. They cannot be executed or changed in the code.
```

Comment:
Currently, the Macros are only preserved during import and export. They cannot be executed or changed in the code.